app/assets/stylesheets/setup-mode.scss

```diff
@@ -17,3 +17,21 @@ body.default-header {
   flex: 1;
   background: white;
 }
+
+.logo-selection {
+  max-height: 80px;
+  height: 80px;
+
+  .required {
+    position: absolute;
+    left: 0;
+    bottom: calc(- var(--fs-xs));
+    font-size: var(--fs-s);
+    color: var(--accent-danger-darker);
+    display: none;
+  }
+
+  :focus:invalid + .required {
```

Comment:
This is using the [:focus](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus) pseudo-class along with the [:invalid](https://developer.mozilla.org/en-US/docs/Web/CSS/:invalid) pseudo-class, so that we can handle the validation message visibility without JavaScript.

https://user-images.githubusercontent.com/833231/142290416-855bbc4c-dcef-400a-9fdc-aa2466ddf34b.mp4

Reply:
One thing I feel is a bit off with this, is when you click the 'Upload Logo' button, there's kind of an odd flash of this 'Logo is required' text:

https://user-images.githubusercontent.com/20773163/142381576-c1739812-72be-4db2-a625-f01333dfec76.mp4

Also I noticed the padding on the right of the button seems a bit off (maybe the message is causing it to stretch out a bit?)

Reply:
Hey, random idea/thought... And it might be too late for this now... Buuuut gonna say it anyway :D... Do we need to style this input at all? It clearly causes some troubles and a need to do workarounds with label and all kinds of pseudo classes etc... What if we just kept standard style for `input[file]` and show logo preview below it and that's it? I know it won't be that aesthetically pleasing anymore but at least it will be much simpler and it will rely on pretty standard UX patterns.

This is also a thought after all that [conversation around buttons and CTAs](https://forem.team/nickytonline/buttons-links-and-accessbility-587p) too.

Reply:
That's a good point @ludwiczakpawel. The styling was a separate issue but I had styled it so that the preview would fit better and also so the button would look better.

@Ridhwana, are you OK with me unstyling the button for now and we can discuss this further with design as it's more of a application wide consideration?

Reply:
Sounds good to me! 👍🏽

Reply:
Cool! Apologies from design side: I could probably save you some of the wasted time and several grey hairs if I thought it through more thoroughly earlier. I didn't think it would be that tricky.

Reply:
No worries. I've put it back to default styling. You can see it in action in the PR description QA section.